File: solid-principles/python/python-dip-notification/Alert.py
from Mail import Mail
from Notification import Notification
from User import User

class Alert:
    def __init__(self):
        self.communication_list = []
        self.mail = Mail()
        self.notification = Notification()
        self.communication_list.append(self.mail)
        self.communication_list.append(self.notification)

    def notify(self, user, message):
        for communication in self.communication_list:
            communication.send_info(user, message)

File: solid-principles/csharp/csharp-dip-notification/Alert.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp_haber
{
    public class Alert
    {
        private Mail mail;
        private Notification notification;
        private List<Communication> comList;

        public Alert(){
            comList = new List<Communication>();
            mail = new Mail();
            notification = new Notification();
            comList.Add(mail);
            comList.Add(notification);

        }

        public void Notify(User user, String message){
            foreach (Communication communication in comList){
                communication.SendInfo(user, message);
            }
        }
    }
}